src/kernels/raw/launcher/kernelProcess.node.ts

```diff
@@ -584,25 +587,20 @@ export class KernelProcess implements IKernelProcess {
             // First part of argument is always the executable.
             const executable = this.launchKernelSpec.argv[0];
             traceInfo(`Launching Raw Kernel ${this.launchKernelSpec.display_name} # ${executable}`);
-            const promiseCancellation = createPromiseFromCancellation({ token: cancelToken, cancelAction: 'reject' });
             const [executionService, env] = await Promise.all([
-                Promise.race([
-                    this.processExecutionFactory.create(this.resource),
-                    promiseCancellation as Promise<IProcessService>
-                ]),
+                this.processExecutionFactory.create(this.resource, cancelToken),
                 // If we have an interpreter always use that, its possible we are launching a kernel that is associated with a Python environment
                 // E.g. we could be dealing with a Java/R kernel registered in a conda environment.
                 // Note that there might still be python env vars to merge from the kernel spec in the case of something like
                 // a Java kernel registered in a conda environment
-                Promise.race([
```

Comment:
This the debt, we are cancelling a promise instead of passing a cancellation token to the code that returns the promise.
Basically now we do not have a promise.race